src/main/java/nollid/commands/EventCommand.java

```diff
@@ -0,0 +1,149 @@
+package nollid.commands;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+import nollid.Parser;
+import nollid.Storage;
+import nollid.TaskList;
+import nollid.exceptions.InvalidArgumentException;
+import nollid.exceptions.NollidException;
+import nollid.tasks.Event;
+
+/**
+ * EventCommand class represents a command for adding an Event task.
+ * It extends the Command class and implements the execute method to perform the command logic.
+ */
+public class EventCommand extends Command {
+    /**
+     * Constant string providing usage hint for the EventCommand.
+     */
+    public static final String USAGE_HINT =
+            "Usage: event [task description] /from [d/m/yyyy] {hh:mm 24hr format} " + "/to [d/m/yyyy] {hh:mm 24hr "
+                    + "format}";
+
+    /**
+     * ArrayList containing command arguments.
+     */
+    private final ArrayList<String> argsList;
+
+    /**
+     * Constructor for EventCommand.
+     *
+     * @param argsList ArrayList containing command arguments.
+     */
+    public EventCommand(ArrayList<String> argsList) {
+        this.argsList = argsList;
+    }
+
+    /**
+     * Overrides the execute method from the Command class.
+     * Executes the command to add an event task.
+     */
+    @Override
+    public String execute(TaskList tasks, Storage storage) throws NollidException {
+        int fromIndex = this.argsList.indexOf("/from");
+        int toIndex = this.argsList.indexOf("/to");
+
+        checkDescriptionNotEmpty(fromIndex, toIndex);
+        checkStartNotEmpty(fromIndex, toIndex);
+        checkEndNotEmpty(fromIndex, toIndex);
+
+        StringBuilder taskDescription = new StringBuilder();
+        StringBuilder from = new StringBuilder();
+        StringBuilder to = new StringBuilder();
+
+        // Deal with the user sending "/from" before "/to" or vice versa
+        if (fromIndex < toIndex) {
+            extractEventInfo(this.argsList, fromIndex, toIndex, taskDescription, from, to);
+        } else {
+            extractEventInfo(this.argsList, toIndex, fromIndex, taskDescription, to, from);
+        }
+
+        try {
+            LocalDateTime fromDateTime = Parser.getLocalDateTimeFromString(from.toString());
+            LocalDateTime toDateTime = Parser.getLocalDateTimeFromString(to.toString());
+            Event task = new Event(taskDescription.toString(), fromDateTime, toDateTime);
+            tasks.add(task);
```

Comment:
LGTM in terms of catching the error, but if I recall, the try block should be as specific and small as possible, ideally encompassing the lines only the code that will throw the exception you are attempting to catch.

Reply:
That's a great point. Fixed in 8ff138f519ba88b6abb6e03d1dd10b9079a1ec11